### PR TITLE
Get the correct setup fee limit from SharedPreferences

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
- Adds "flutter." prefix to SharedPreferences key for setup fee limit on extensions
  - This is the [default prefix](https://github.com/flutter/packages/blob/b21c5424783a9e74cb841a8df7a15f26eba04a7b/packages/shared_preferences/shared_preferences/lib/shared_preferences.dart#L18) used by shared_preferences package

### CODE REVIEW

File: android/app/src/main/kotlin/com/cBreez/client/BreezForegroundService.kt [hunk 1]
```diff
@@ -39,8 +39,8 @@ class BreezForegroundService : ForegroundService() {
                 "ERROR" -> Logger.tag(TAG).error { l.line }
                 "WARN" -> Logger.tag(TAG).warn { l.line }
                 "INFO" -> Logger.tag(TAG).info { l.line }
-                "DEBUG" -> Logger.tag(TAG).debug { l.line }
-                "TRACE" -> Logger.tag(TAG).trace { l.line }
+                // "DEBUG" -> Logger.tag(TAG).debug { l.line }
+                // "TRACE" -> Logger.tag(TAG).trace { l.line }
```
Comment (dangeross) [inline android/app/src/main/kotlin/com/cBreez/client/BreezForegroundService.kt:43]: Do these need reseting?

Comment (erdemyerebasmaz) [reply android/app/src/main/kotlin/com/cBreez/client/BreezForegroundService.kt]: No, it's continuation of
- #817 
but I'll address it as part of another PR.

**Comment by erdemyerebasmaz** [comment]
There's [an issue on iOS](https://stackoverflow.com/a/76355960) where we can't access shared_preferences values from the extension. Addressing this by using [shared preferences with App Group](https://pub.dev/packages/shared_preference_app_group) plugin when setting channel setup fee limit.

File: lib/utils/preferences.dart [hunk 2]
```diff
@@ -65,6 +66,8 @@ class Preferences {
     _log.info("set payment options channel setup limit fee : $channelFeeLimitMsat");
     final prefs = await SharedPreferences.getInstance();
     await prefs.setInt(_kPaymentOptionChannelSetupFeeLimit, channelFeeLimitMsat);
+    // iOS Extension requirement
+    await SharedPreferenceAppGroup.setInt(_kPaymentOptionChannelSetupFeeLimit, channelFeeLimitMsat);
```
Comment (dangeross) [inline lib/utils/preferences.dart:70]: Is there a need to set it twice? We could also get the preference in the same way using `SharedPreferenceAppGroup.get(_kPaymentOptionChannelSetupFeeLimit)`

Comment (dangeross) [reply lib/utils/preferences.dart]: Makes sense now sorry, it's iOS only

**Review by dangeross — APPROVED** [review]
LGTM

**Comment by erdemyerebasmaz** [comment]
- Refactored PaymentOptionsPage & PaymentOptionsBloc as there were a lot of unwanted behaviors due to bad state management practices.
- Removed PaymentOptionsBloc tests temporarily.

File: lib/utils/preferences.dart (hunk 2 above)
Comment (erdemyerebasmaz) [reply lib/utils/preferences.dart]: Yes, this is workaround for not being able to access `shared_preferences` values on iOS extension. Values set through `shared_preference_app_group` live on app group's UserDefaults and do not overlap with `shared_preferences` values.

**Pull request merged**